\begin{lstlisting}[language=perl]
#--------------------------------------------------
# # Orthograph config file
#-------------------------------------------------- 
#
# Comments start with a hash sign (#) and are ignored by the parser.
# This way you can comment out lines you don't need, but would like to keep
# in your config file because they might be important later.
#
# Settings in UPPER-CASE must be changed to your local environment. Settings
# in lower-case are sane defaults.

#--------------------------------------------------
# # Important stuff first. These settings are mandatory.
#-------------------------------------------------- 

#
# # MySQL connection settings.
#
mysql-username     = USERNAME
mysql-password     = PASSWORD
mysql-database     = DATABASE

# The path to your transcriptome data file. It is always best to use absolute
# paths. You may only supply one file, not a directory of files.
estfile            = FASTAFILE

# Species name. Make sure to pick a unique name because otherwise results get
# mixed up. You can, however, supply an existing name if you want to add
# sequences to an existing data set. 
species-name       = TEST-SPECIES

# The ortholog set you created earlier.
ortholog-set       = SETNAME

# Reference taxa. Must be a comma-separated list of taxa that are present in
# your ortholog set. If you don't specify any reference taxa (\ie, comment out
# this setting), all taxa in your set are used as reference taxa. 
reference-taxa     = COMMA, SEPARATED, LIST, OF, TAXA, IN, YOUR, ORTHOLOG, SET

# This is where your results are placed. Will be created if it doesn't exist.
output-directory   = /PATH/TO/OUTPUT-DIR


#--------------------------------------------------
# Options. These settings are optional, but may be required if the defaults
# don't do.
#-------------------------------------------------- 

#
# # Paths to the programs. It's best to set absolute paths.
#
# Default alignment program; used to create the ortholog set. Must accept a
# fasta input file as input and produce fasta-formatted output on STDOUT. 
# Note: The --anysymbol option makes MAFFT accept any character in a sequence,
# INCLUDING '*' for stop codons and 'U' for Selenocystein. If you are not OK
# with this, you may remove the --anysymbol option from the command, but then it
# is your responsibility to make sure that your ortholog set sequences do not
# contain any nonstandard symbols that may make MAFFT choke on your set.
#
# Standard amino acid symbols are: ACDEFGHIKLMNPQRSTVWY and X for ambiguity.
alignment-program    = mafft --localpair --maxiterate 1000 --anysymbol

# HMMbuild is used to build the profile HMMs. Part of the HMMER3 package.
hmmbuild-program     = hmmbuild

# makeblastdb is used to build the BLAST database. Part of NCBI BLAST+
makeblastdb-program  = makeblastdb

# Fastatranslate is part of the Exonerate package and translates the transcript
# sequences into all six reading frames. You are free to use different
# programs, though.
translate-program    = fastatranslate

# hmmsearch, of course. Also part of the HMMER3 package.
hmmsearch-program    = hmmsearch

# BLAST. Should be blastp from the NCBI BLAST+ package.
blast-program        = blastp

# Exonerate. Not used yet.
#exonerate-program   = exonerate

#
# # MySQL settings
#
# The database server. Change this if the database does not run on the same
# computer as the analysis. Ask your administrator if you don't know what to
# write here.
# Defaults to 127.0.0.1 (localhost).
#mysql-dbserver               = 127.0.0.1

# Prefix for your Orthograph database tables. Useful if you are running
# multiple instances of Orthograph on the same database but don't want the data
# to be mixed up. Defaults to 'orthograph'.
#mysql-prefix                 = orthograph

#
# # Settings that affect the HMM and BLAST searches.
#
# E-value threshold. The hmmsearch e-value threshold affects the specificity of
# the HMM search, the first step in the reciprocal algorithm. It defines how
# distantly related candidate orthologs may be when searching through the
# transcriptome file. The BLAST e-value threshold affects the second step, the
# reciprocal search. Basically, this defines the false-positive probability
# (lower e-value = lower probability).
hmmsearch-evalue-threshold    = 1e-05
blast-evalue-threshold        = 1e-05

# You can also set a score threshold. A higher score means a better match. For
# hmmsearch, E-value and score thresholds are mutually exclusive. If both are
# set, the E-value threshold will be used (to use the score threshold, you have
# to unset the E-value threshold). BLAST accepts both E-value and score
# thresholds simultaneously.
#hmmsearch-score-threshold	  = 10
#blast-score-threshold        = 10

# Maximum number of hmmsearch hits to consider. This setting is useful to limit
# the number of reciprocal searches for very large number of hmmsearch hits.
# However, it is also useful to limit the scope of your HMM searches, which you
# normally don't want to do. Unless you have an idea of how many reciprocal
# searches are required to effectively verify or reject a candidate ortholog,
# don't change this setting. 
# Defaults to 1000.
#max-blast-searches           = 1000

# The soft threshold is a special concept of Orthograph: While checking in
# strict mode (see below) whether the reciprocal hits are part of the ortholog
# group in question, this many mismatches may occur before the ortholog
# candidate is rejected.
# Defaults to 5.
#soft-threshold               = 5

# Maximum number of BLAST hits to consider for ortholog candidates.
# Defaults to 100.
blast-max-hits                = 100

# 
# # Other options
#
# Strict search. Normally it is enough for a match to occur if one of the
# reference taxa is hit in the reciprocal search. In strict mode ALL reference
# taxa must be hit to verify an ortholog assignment. This is (much) more
# conservative.
#strict                       = 1

# Clear pre-existing data of the same species from the database prior to the
# analysis. Recommended if you plan to run the same analysis multiple times,
# but doesn't hurt otherwise. 
# This is the default, uncomment this line if you want to turn it off.
#clear-results-from-database  = 0

# Delete old result files. This means the hmmsearch and BLAST report files found
# in the output directory. If you plan to run the same analysis multiple times
# with the same hmmsearch and BLAST settings, then don't have them deleted. This
# will speed up the process significantly, since the search programs don't have
# to be run again.
# This is the default. Uncomment this line if you want the files deleted.
#clear-result-files           = 1

# Selenocysteine (U) may occur in some protein sequences. However, some
# alignment programs do not accept this nonstandard amino acid symbol. You can
# tell Orthograph to substitute all 'U' in the sequences with a different
# character. The default is not to substitute.
#substitute-u-with            = X

# Verbose output. More information about the hmmsearch and BLAST hits. Normally
# you don't want to see this. If you are really interested in what Orthograph is
# thinking during the analysis, uncomment this. Verbose and quiet are mutually
# exclusive.
#verbose                      = 1

# Quiet output. Uncomment this if you don't want to be bothered during the
# analysis. Verbose and quiet are mutually exclusive.
#quiet                        = 1


#
# # More paths
# 
# Sets directory. Useful if you would like to keep your ortholog sets (that is,
# the BLAST database, the HMMs and the alignment files for each ortholog gene)
# in a separate place. Defaults to 'sets' (in the current directory).
#sets-dir                     = /PATH/TO/SETS/DIR

# Log file. If set, all messages will also be written to this file. If this is
# not set, messages are not written to a log file, but to STDOUT and STDERR.
#logfile                      = /PATH/TO/LOGFILE
\end{lstlisting}
