%\RequirePackage[l2tabu, orthodox]{nag}
\documentclass[a4paper,12pt]{scrreprt}
\usepackage[utf8]{inputenc}   % input encoding
\usepackage[T1]{fontenc}      % font encoding
\usepackage[english]{babel}		% English language 
\usepackage{lmodern}					% font looks better on screen
\usepackage{microtype}				% improve kerning
% if using natbib
\usepackage{natbib}           % bibliography for scientists
% if using biblatex
%\usepackage[backend=bibtex,style=authoryear,bibstyle=nature,natbib=true]{biblatex}
%\usepackage{csquotes}         % dunno what this does
%--------------------------------------------------
% more features
\usepackage{amsmath}					% all the fancy math stuff
\usepackage{caption}          % customize captions
\usepackage{color}						% enable color
%\usepackage{enumerate}				% customize numbered lists
\usepackage{graphicx}					% fancy graphics
\usepackage{listings}         % code listings
\usepackage{tabularx}         % extended tables
%\usepackage{tikz}             % draw stuff with tikz
\usepackage{url}							% URL handling
\usepackage{xspace}						% to space or not to space
\usepackage{hyperref} 				% should be loaded last
%--------------------------------------------------

% new colors and commands
\definecolor{lightgrey}{rgb}{0.9,0.9,0.9}
\newcommand{\file}[1]{{\lstinline{#1}}}
\newcommand{\code}[1]{{\texttt{#1}}}
\newcommand{\hamstr}{HaMStR\xspace}
\newcommand{\pname}{Orthograph\xspace}
\newcommand{\pfullname}{\emph{Ortho}logy prediction using a \emph{gr}aph-based \emph{a}pproach with \emph{p}rofile \emph{h}idden Markov models\xspace}
\newcommand{\myself}{Malte Petersen}
\newcommand{\mytitle}{Fast and efficient mapping of transcript sequences to ortholog groups}
\newcommand{\species}[1]{\textit{#1}}
\newcommand{\ie}[1]{\textit{i.e.,}}
\newcommand{\eg}[1]{\textit{e.g.,}}
\newcommand{\todo}[1]{\textbf{\color{red}[#1]}}

% setup biblatex
%--------------------------------------------------
% \bibliography{bib/diploma}
% \ExecuteBibliographyOptions{
% 	firstinits=true,
% 	backref=true,
% 	isbn=false,
% 	url=false,
% 	maxcitenames=2,
% 	maxbibnames=999,
% }
% \renewcommand{\bibfont}{\normalfont\footnotesize}
%-------------------------------------------------- 

% setup natbib citation format
\bibpunct{(}{)}{,}{a}{,}{,}

% setup caption format
\captionsetup{font=small, labelfont=bf}

% setup the listings package
\lstset{%
	language=perl,
	basicstyle=,                       % size of the fonts that are used for the code
	numbers=left,                      % where to put the line numbers
	numberstyle=\tiny,                 % size of the fonts that are used for the line numbers
	%stepnumber=2,                      % step between two line numbers. 
	numbersep=5pt,                     % how far the line numbers are from the code
	backgroundcolor=\color{lightgrey}, % choose the background color. Requires \usepackage{color}
	showspaces=false,                  % show spaces adding particular underscores
	showstringspaces=false,            % underline spaces within strings
	showtabs=false,                    % show tabs 
	frame=none,                        % none|leftline|topline|bottomline|lines|single, shadowbox 
	%frameround=tttt,                   % rounded corners; t=round, f=corner
	tabsize=2,                         % sets default tabsize to 2 spaces
	captionpos=t,                      % caption position (t|b)
	breaklines=true,                   % automatic line breaking
	breakatwhitespace=false,           % automatic breaks should not happen at whitespace
	%escapeinside={\%}{)}               % if you want to add a comment within your code
}

% setup hyperlinks, bookmarks and pdf metadata
\hypersetup{%
	bookmarks=true,
	pdfborder={0 0 0},
	pdftitle={\mytitle},
	pdfauthor={\myself},
	pdfcreator={pdflatex},
	pdfsubject={orthology prediction},
	pdfkeywords={orthology} {prediction} {est} {transcriptome} {hmm} {phylogenetics} {thesis} {1kite}
}
% alleviate compression so pdftk won't complain
\pdfobjcompresslevel=1

% doublespace
\linespread{1.3}

%--------------------------------------------------
% start of actual document
%-------------------------------------------------- 
\begin{document}
\begin{titlepage}
	\include{inc/titlepage}
\end{titlepage}

\pagestyle{empty}
\chapter*{Paper notes}
	\input{inc/notes}

\phantomsection
%\addcontentsline{toc}{chapter}{Declaration of authorship}
\chapter*{Declaration of authorship}
	\input{inc/declaration}
	\clearpage

\pagestyle{headings}
\pagenumbering{roman}

\tableofcontents
\clearpage

\phantomsection
\addcontentsline{toc}{chapter}{List of figures}
\listoffigures
\clearpage

\phantomsection
\addcontentsline{toc}{chapter}{List of tables}
\listoftables
\clearpage

\phantomsection
\include{inc/quotes}

\pagenumbering{arabic}

\chapter{Introduction}
	\input{inc/intro/intro}
	\section{Orthology}
		\input{inc/intro/orthology}
	\section{How can orthology be assessed?}
		\input{inc/intro/orthology-howto}
	\section{Transcriptomes and additional challenges}
		\input{inc/intro/transcriptomes}
	\section{\hamstr}
		\input{inc/intro/hamstr}
		\subsection{Hidden Markov models}
			\input{inc/intro/hmms}
			\subsubsection{Sequence weighting in profile HMMs}
				\input{inc/intro/sequence-weighting}
		\subsection{Room for improvements in \hamstr}
			\input{inc/intro/improvements}

\chapter{Material and methods}
	\section{Hardware}
		\input{inc/methods/hardware}
	\section{Programs}
		\input{inc/methods/programs}
		\subsection{HMMER3}
			\input{inc/methods/hmmer}
		\subsection{BLAST+}
			\input{inc/methods/blast}
		\subsection{Exonerate}
			\input{inc/methods/exonerate}
		\subsection{MySQL}
			\input{inc/methods/mysql}
	\section{Perl programming}
		\input{inc/methods/perl}
	\section{Tools}
		\input{inc/methods/tools}

\chapter{Results}
	\input{inc/results/intro}
	\section{Graph}
		\input{inc/results/graph}
	\section{Database structure}
		\input{inc/results/database-structure}
	\section{Algorithm}
		\input{inc/results/algorithm}
		\subsection{Analysis}
			\input{inc/results/algorithm-analysis}
		\subsection{Reporting}
			\input{inc/results/algorithm-reporting}
	\section{Implementation decisions}
		\subsubsection{Traversing the hit list by e-value}
			\input{inc/results/traversing-by-evalue}
	\section{Performance tweaks}
		\subsection{One BLAST database comprising all core taxa proteomes}
			\input{inc/results/one-blast}
		\subsection{RBDMS vs. flat files}
		\subsection{MySQL performance}
		\subsubsection{Individual tables per query species}
			\input{inc/results/table-per-species}
	\section{SHA-256 checksums as sequence identifiers}
		\input{inc/results/checksums}
	\section{Benchmarks}

\chapter{Discussion and outlook}
	\input{inc/discussion/intro}
	\section{Conclusion and outlook}
		\subsection{Future development}
		\input{inc/discussion/future-devel}
	\section{Summary}

\phantomsection
\addcontentsline{toc}{chapter}{Acknowledgements}
\chapter*{Acknowledgements}
	\input{inc/acknowledgements}
	\clearpage

\phantomsection
\addcontentsline{toc}{chapter}{Bibliography}
% uncomment these four lines if using bibtex
\footnotesize
\bibliographystyle{plainnat}
\bibliography{bib/diploma}
\normalsize
% uncomment this line if using biblatex
%\printbibliography
\clearpage

\phantomsection
\appendix
\chapter{Appendix}
	\section{Listings}
		\input{inc/appendix/listings}
	\section{Example configuration file}
		\input{inc/appendix/example-config}

\end{document}
