\hamstr implements a graph-based approach using profile hidden Markov models
(HMMs, explained in \autoref{sec:hmms}). It is aimed specifically at
searching for orthologs in \emph{expressed sequence tag} (EST)
data\footnote{National Center for Biotechnology Information: EST fact sheet
(\url{http://www.ncbi.nlm.nih.gov/About/primer/est.html}, revised: March 29,
2004)}, which are sequenced using complementary DNA (cDNA) libraries in a
process called \emph{shotgun sequencing}. The cDNA is cloned from mRNA and
therefore contains no introns, and, like a transcriptome, EST data is
incomplete, i.e., it contains only nucleotide sequences of the genes that were
expressed at the time of RNA preservation. Due to the way it is sequenced and
assembled, EST data can be redundant and fragmented as well, which is why
methods for orthology prediction in genomic data cannot be applied, and
approaches like described above have to be used.

\hamstr cannot be used to perform \emph{de novo} orthology assessment. Instead,
it relies on pre-computed ortholog goups that data from the query EST library is
clustered to. These so-called \emph{core orthologs} are compiled from existing
databases such as OrthoDB and must be of species for which a complete genome is
available. From each \emph{ortholog group} (OG, a set of genes that are known to
be one-to-one orthologous), a HMM is generated: each HMM is a probabilistic
representation of the amino acid sequence of a gene. 

\hamstr uses a three-step triangulation strategy as depicted in
\autoref{fig:hamstr-triangulation}: the EST sequence space is searched with the
HMMs for matches (\autoref{fig:hamstr-triangulation}a). To verify orthology in
the HMM hits, each hit sequence is compared via BLASTP \citep{altschul1997} to a
so-called reference taxon (\autoref{fig:hamstr-triangulation}b). Note that the
proteome, itself being complete, contains the amino acid sequences that are
known to be orthologous. If the BLAST search hits a sequence that was used to
generate the HMM (\autoref{fig:hamstr-triangulation}c), then orthology is
assumed and the EST in question is added to the ortholog cluster. Otherwise, it
is discarded.

\input{inc/intro/fig-hamstr-triangulation}

% enable this if you need it, it makes compilation a bit slower
%\input{inc/intro/hamstr-flowchart}

\todo{more elaboration here}
