Phylogenetic analysis has always required orthology among investigated genes.
With next-generation, high-throughput sequencing technologies providing vast
amounts of data, comparing genomes is becoming a more and more important task. 

To make use of the speciation information contained in the ortholog property, it
is important to classify conserved genes according to their homologous
relationships. It is worth mentioning here that homology is a concept of
quality, not quantity (\cite{reeck1987}) and thus indivisible. Two sequences can
be \emph{similar} by a percentage (e.g., amino acid positional identity), but
they are either homologous or they are not. The same follows for orthologs,
paralogs, and xenologs. This distinction is important because homology implies a
genealogical relationship, whereas similarity does not. Similarity can also be
the result of other evolutionary processes, such as convergence, which results
in \emph{analogy}. All variants of similarity can be grouped under the term
\emph{homoplasy}, which encompasses similarity (\emph{homos} (``equal''),
\emph{plasis} (``shaping'')) excluding homology and its subforms \cite{page1998}.

A simple alignment or scoring (i.e., similarity) alone cannot separate
homologous from merely similar sequences (\cite{eisen1998}). Since orthology and
paralogy are notions that describe evolutionary events, they can be directly
coupled to a species tree topology. This is how \emph{tree reconciliation
strategies} work: a gene tree topology is compared to the topology of the chosen
species tree. The two trees are reconciled using a maximum parsimony criterion,
i.e., the minimal possible number of evolutionary events (speciation, gene
duplication, gene loss). In order to produce a reconciled tree, two major
approaches are followed: \emph{consensus} and \emph{approximation} (figure
\ref{fig:consens-approx}). Like all algorithms in this class, they consider a
phylogenetic tree a formal mathematical object. The consensus algorithm tries to
find a tree that combines the features common to both of the original trees.
This is rarely a well-resolved tree and therefore biologically not very
relevant, as the consensus criterion leads to clustering (compare figure
\ref{fig:consens-approx}c). More initial trees reduce the number of common
clusters, and the concept of a ``common cluster'' can be altered, e.g., to
include only those clusters that occur in 80\% of the initial trees, but in most
cases, clustering cannot be avoided altogether (\cite{mirkin1995}).

In the approximation approach, 
something in \cite{page1997}

\input{inc/intro/fig-consens-approx}

See figure \ref{fig:hamstr}.
