\label{sec:algorithm-reporting}
The evaluation program \tool{orthograph-reporter} assesses ortholog
relationships among the transcript sequence data and the known groups of
orthologous amino acid or nucleotide sequences in the database on the basis of
the bidirectional search results that were obtained using
\tool{orthograph-analyzer}. Note that this step is intended to be independent of
the search algorithm and does not modify the results that are already present in
the database, leaving them in a consistent state for evaluation under different
criteria.

For generating the graph, \tool{orthograph-reporter} organizes the information
from the database in a multidimensional data structure that is easy to parse:

\begin{verbatim}
hmmsearch_evalue => {
  ortholog_group_id => [
    reciprocal_hit,
    reciprocal_hit,
  ],
  ortholog_group_id => [
    reciprocal_hit,
    reciprocal_hit,
  ],
  . . .
},
. . .
\end{verbatim}

Each \emph{reciprocal\_hit} represents a BLAST search against the database of
reference proteomes for the ortholog set that was used. The query amino acid
sequence was obtained during the preceding HMM search. The BLAST search hit can
be compared to a complete list of amino acid sequences that were used in
constructing the HMM for the given ortholog group. If the reciprocal search has
hit a matching amino acid sequence, this pair of [ortholog group + transcript
sequence] is considered orthologous.

The reporting algorithm is as follows (see also \autoref{fig:orthograph-graph}):

\begin{enumerate}
	\item Get configuration, initialize global variables.
	\item Check whether the configuration is acceptable (see algorithm for
		\tool{orthograph-analyzer}).
	\item Get a list of ortholog group IDs and their associated ortholog sequences
		from the database.
	\item Get all search results from the database.
	\item Sort the \tool{hmmsearch} e-values in ascending order.
	\item Starting with the lowest \tool{hmmsearch} e-value, do the following:
	\begin{enumerate}
		\item For each ortholog group that has obtained a \tool{hmmsearch} hit with this
			e-value, do the following:
			\begin{enumerate}
				\item Sort the reciprocal hits by their BLAST e-value in ascending order.
				\item For each reciprocal hit, do the following:
				\begin{enumerate}
					\item If the reciprocal search for this transcript has hit a sequence
						that belongs to this ortholog group, and this transcript has not been
						assigned previously, then this pair of [ortholog group + transcript
						sequence] is assumed to be a valid match.  Continue. Otherwise, skip
						to the next reciprocal hit unless the ``soft threshold'' has been
						reached. In the latter case, skip to the next ortholog group.
					\item If the transcript does not overlap with an existing assignment,
						assign this transcript to this ortholog group. This transcript cannot
						be assigned again.
				\end{enumerate}
			\item If there is a gap between the transcripts (the end of one transcript
				lies more than 1 bp before the start of the next), fill the gap with
				'X' and concatenate the fragments. This ortholog group can also not be
				assigned again.
			\item Correct for frameshift errors and infer the coding sequence using
				Exonerate (not yet implemented at the time of this writing).
		\end{enumerate}
	\end{enumerate}
\end{enumerate}

The results are written to output files---one for each ortholog group with the
transcript sequences that were mapped to it---and a summary table. The presence
of protein-coding nucleotide sequence data for the reference proteomes in the
database allows output of the corresponding nucleotide sequences along with the
amino acid sequences.

\input{inc/results/fig-orthograph-graph}

