The database was designed in a way that reduces redundant information to a
minimum. The structure diagram is shown in Figure \ref{fig:dbstructure} (page
\pageref{fig:dbstructure}). All tables except the ``ests'', ``hmmsearch'', and
``blast'' tables are used to store information about the amino acid or
nucleotide sequences that belong to either an ortholog set or a proteome of a
core taxon, or both. 

All tables have an ``id'' column with an auto-incremented integer as a unique
identifier. Normally, this identifier is used in \code{JOIN} queries that
reference other tables. The tables ``aaseqs'' and ``ntseqs'' contain the
peptide or nucleotide sequences, respectively, that belong to the ortholog sets.
One (or both, if nucleotide sequences are present) are referenced to a sequence
pair in the table ``sequence\_pairs''. The table ``set\_details'' contains a
shorthand name for each ortholog set along with its longer name or description. 


% this is probably not that useful
%\input{inc/results/tab-mysql-variables}

\input{inc/results/fig-dbstructure}
